pkg/provider/azure_loadbalancer_test.go

```diff
@@ -161,32 +162,35 @@ func TestGetLoadBalancer(t *testing.T) {
 			},
 		},
 	}
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
 
-	for i, c := range tests {
-		az := GetTestCloud(ctrl)
-		mockPIPsClient := az.PublicIPAddressesClient.(*mockpublicipclient.MockInterface)
-		if c.pipExists {
-			mockPIPsClient.EXPECT().List(gomock.Any(), "rg").Return([]network.PublicIPAddress{
-				{
-					Name: pointer.String("testCluster-aservice"),
-					PublicIPAddressPropertiesFormat: &network.PublicIPAddressPropertiesFormat{
-						IPAddress: pointer.String("1.2.3.4"),
-					},
-				},
-			}, nil)
-		} else {
-			mockPIPsClient.EXPECT().List(gomock.Any(), "rg").Return([]network.PublicIPAddress{}, nil).MaxTimes(2)
-		}
-		mockLBsClient := az.LoadBalancerClient.(*mockloadbalancerclient.MockInterface)
-		mockLBsClient.EXPECT().List(gomock.Any(), az.Config.ResourceGroup).Return(c.existingLBs, nil)
+	for _, c := range tests {
```

Comment:
Cover dualstack scenario in this test?

Reply:
Thanks for the suggestion.

Yes, dualstack should be covered in this test but it needs code change in those frontend IP config related methods like ``getServiceLoadBalancerStatus()`. It should return 2 `fipConfig`.

Considering this PR is large enough, I plan to add those changes in the next PR. wdyt?

Reply:
I mean we should cover the changes of GetLoadBalancer in this PR, which is to test ipv4 and ipv6 pips at a time.

Reply:
Discussed offline. Decided to cover dualstack for `existsPip()` part.